connect/transforms/src/test/java/org/apache/kafka/connect/transforms/CastTest.java

```diff
@@ -320,6 +324,68 @@ public void castWholeRecordValueSchemalessUnsupportedType() {
         xformValue.apply(new SourceRecord(null, null, "topic", 0, null, Collections.singletonList("foo")));
     }
 
+    @Test
+    public void castLogicalToPrimitive() {
+        List<String> specParts = Arrays.asList(
+            "date_to_int32:int32",  // Cast to underlying representation
+            "timestamp_to_int64:int64",  // Cast to underlying representation
+            "time_to_int64:int64",  // Cast to wider datatype than underlying representation
+            "decimal_to_int32:int32",  // Cast to narrower datatype with data loss
+            "timestamp_to_float64:float64",  // loss of precision casting to double
+            "date_to_string:string",  // cast to string representation
+            "decimal_to_string:string",  // cast to string representation
+            "null_value:int32"
```

Comment:
Nit: this is actually a timestamp column with a null value, and it might be worth calling that out here.
```suggestion
            "null_timestamp_to_int32:int32"
```